data/google-tag-manager.json

```
@@ -15,9 +15,10 @@
         "key": "gtm"
       },
       {
-        "code": "window[{{l}}]=window[{{l}}]||[];window[{{l}}].push({'gtm.start':new Date().getTime(),event:'gtm.js'});",
+        "code": "window[{{l}}]=window[{{l}}]||[];{{#consentValues}}(function () {window[{{l}}].push(arguments)})('consent', {{consentType}},{{consentValues}});{{/consentValues}}window[{{l}}].push({'gtm.start':new Date().getTime(),event:'gtm.js'});",
         "optionalParams": {
-          "l": "dataLayer"
+          "l": "dataLayer",
+          "consentType": "default"
```

I think `consentValues` needs to be present here, like with Analytics? https://github.com/GoogleChromeLabs/third-party-capital/blob/main/data/google-analytics.json#L23
```suggestion
          "consentType": "default",
          "consentValues": null
```

---

oh, consentValues can be undefined. null and undefined both count as falsy boolean expression. We can also remove it from analytics so the JSON can be slightly lighter.

---

> oh, consentValues can be undefined. null and undefined both count as falsy boolean expression. We can also remove it from analytics so the JSON can be slightly lighter.

Generally makes sense, however wouldn't it be more intuitive to still have those be explicitly listed in optionalParams? I think it helps DX to be able to quickly scan the params and optionalParams and know what's available. Otherwise you'll have to look at the less obvious and harder to parse "code" value.

I don't think it technically makes a difference whether they're explicitly defined or not, but personally I think there's a benefit of keeping them in there.

---

Does PHP unit test needs some changes ?

---

I get it now, llet's keep it :+1: